src/mds/MDCache.cc

```diff
@@ -3083,6 +3083,7 @@ void MDCache::handle_resolve(MMDSResolve *m)
 
   if (mds->get_state() < MDSMap::STATE_RESOLVE) {
     if (mds->get_want_state() == CEPH_MDS_STATE_RESOLVE) {
+      discard_delayed_resolve(from);
```

Comment:
delayed_resolve is only inserted to further down in this function, so I don't see how it's possible for it to already contain entries in the case where "mds->get_state() < MDSMap::STATE_RESOLVE".  Something I'm missing?